planner/core/testdata/plan_suite_in.json

```diff
@@ -572,6 +572,19 @@
       "select /*+ LIMIT_TO_COP() */ a from tn where a > 10 limit 1"
     ]
   },
+  {
+    "name": "TestCTEMergeHint",
+    "cases": [
+      "with cte as (select /*+ MERGE()*/ * from tc where tc.a < 60) select * from cte where cte.a <18",
+      "with cte as (select * from tc where tc.a < 60) select * from cte where cte.a <18",
+      "WITH cte1 AS (SELECT /*+ MERGE()*/ a FROM tc), cte2 AS (SELECT /*+ MERGE()*/ c FROM te) SELECT * FROM cte1 JOIN cte2 WHERE cte1.a = cte2.c;",
+      "WITH cte1 AS (SELECT a FROM tc), cte2 AS (SELECT /*+ MERGE()*/ c FROM te) SELECT * FROM cte1 JOIN cte2 WHERE cte1.a = cte2.c;",
+      "with recursive cte1(c1) as (select 1 union select /*+ merge */ c1 + 1 c1 from cte1 where c1 < 100) select * from cte1;",
+      "with cte1 as (select * from tc), cte2 as (with cte3 as (select /*+ MERGE() */ * from te) ,cte4 as (select * from tc) select * from cte3,cte4) select * from cte2;",
+      "with cte2 as (with cte4 as (select * from tc) select * from te, cte4) select * from cte2;",
+      "with cte2 as (with cte4 as (select /*+ merge() */  * from tc) select * from te, cte4) select * from cte2;"
```

Comment:
Add one case that the hint is failed to apply and raise warning in the output.

Reply:
add warnings for the recursive case in the output